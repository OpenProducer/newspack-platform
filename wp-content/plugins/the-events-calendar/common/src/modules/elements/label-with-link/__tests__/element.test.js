/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import LabelWithLink from '../element';

describe( 'Label With Link Element', () => {
	it( 'renders a label with link', () => {
		const component = renderer.create(
			<LabelWithLink linkHref="#" linkText="test-text" />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders a label with link with class', () => {
		const component = renderer.create(
			<LabelWithLink linkHref="#" linkText="test-text" className="test-class" />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders a label with link with label', () => {
		const component = renderer.create(
			<LabelWithLink linkHref="#" linkText="test-text" label="test label" />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
