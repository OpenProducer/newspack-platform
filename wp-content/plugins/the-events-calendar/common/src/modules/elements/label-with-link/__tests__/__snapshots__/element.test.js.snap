// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label With Link Element renders a label with link 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-link"
>
  <span
    className="tribe-editor__labeled-item__label"
  />
  <a
    className="tribe-editor__link tribe-editor__label-with-link__link"
    href="#"
  >
    test-text
  </a>
</div>
`;

exports[`Label With Link Element renders a label with link with class 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-link test-class"
>
  <span
    className="tribe-editor__labeled-item__label"
  />
  <a
    className="tribe-editor__link tribe-editor__label-with-link__link"
    href="#"
  >
    test-text
  </a>
</div>
`;

exports[`Label With Link Element renders a label with link with label 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-link"
>
  <span
    className="tribe-editor__labeled-item__label"
  >
    test label
  </span>
  <a
    className="tribe-editor__link tribe-editor__label-with-link__link"
    href="#"
  >
    test-text
  </a>
</div>
`;
